src/Microsoft.AspNet.Mvc.Core/ForbiddenResult.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http.Authentication;
+using Microsoft.AspNet.Mvc.Logging;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// An <see cref="ActionResult"/> that on execution issues a 403 forbidden response
+    /// if the authentication challenge is unacceptable.
+    /// </summary>
+    public class ForbiddenResult : ActionResult
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="ForbiddenResult"/>.
+        /// </summary>
+        public ForbiddenResult()
+            : this(new string[] { })
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ForbiddenResult"/> with the
+        /// specified authentication scheme.
+        /// </summary>
+        public ForbiddenResult(string authenticationScheme)
+            : this(new[] { authenticationScheme })
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ForbiddenResult"/> with the
+        /// specified authentication schemes.
+        /// </summary>
+        public ForbiddenResult(IList<string> authenticationSchemes)
+            : this(authenticationSchemes, properties: null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ForbiddenResult"/> with the
+        /// specified <paramref name="properties"/>.
+        /// </summary>
+        public ForbiddenResult(AuthenticationProperties properties)
+            : this(new string[] { }, properties)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ForbiddenResult"/> with the
+        /// specified authentication scheme and <paramref name="properties"/>.
+        /// </summary>
+        public ForbiddenResult(string authenticationScheme, AuthenticationProperties properties)
+            : this(new[] { authenticationScheme }, properties)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ForbiddenResult"/> with the
+        /// specified authentication schemes and <paramref name="properties"/>.
+        /// </summary>
+        public ForbiddenResult(IList<string> authenticationSchemes, AuthenticationProperties properties)
+        {
+            AuthenticationSchemes = authenticationSchemes;
+            Properties = properties;
+        }
+
+        /// <summary>
+        /// The list of authentication components that should handle the authentication challenge
```

Comment:
should this say `authentication schemes`? Is there anything easy we can put here to say what this means?